website/server/controllers/api-v3/tasks/groups.js

```diff
@@ -362,9 +365,12 @@ api.approveTask = {
       direction,
     });
 
-    managerPromises.push(task.save());
-    managerPromises.push(assignedUser.save());
-    await Promise.all(managerPromises);
+    let masterTask = await groupTasks.handleSharedCompletion(task);
+    if (masterTask) approvalPromises.push(masterTask.save());
```

Comment:
If you are awaiting above, then you do not need to push the promise because it will be completed already. I don't think we want to run `handleSharedCompletion` in parallel with `task.save`, so we should just remove pushing it, and only await it.